diff --git a/src/embed-agent.sh b/src/embed-agent.sh
index dda8ba76..ad4f8938 100755
--- a/src/embed-agent.sh
+++ b/src/embed-agent.sh
@@ -10,7 +10,7 @@ resource_compiler="$7"
 resource_config="$8"
 
 priv_dir="$output_dir/frida-agent@emb"
-
+custom_script="$output_dir/../../../../frida-core/src/topatch.py"
 case $host_os in
   macos|ios)
     if [ -z "$LIPO" ]; then
@@ -30,6 +30,9 @@ collect_generic_agent ()
   else
     touch "$embedded_agent"
   fi
+  if [ -f "$custom_script" ]; then
+    python3 "$custom_script" "$embedded_agent"
+  fi
   embedded_agents+=("$embedded_agent")
 }
 
@@ -61,7 +64,9 @@ case $host_os in
       echo "An agent must be provided"
       exit 1
     fi
-
+    if [ -f "$custom_script" ]; then
+      python3 "$custom_script" "$embedded_agent"
+    fi
     exec "$resource_compiler" --toolchain=gnu -c "$resource_config" -o "$output_dir/frida-data-agent" "$embedded_agent"
     ;;
   *)
